tests/unit/core/middleware/test_cspMiddleware.js

```diff
@@ -5,13 +5,111 @@ import parse from 'content-security-policy-parser';
+        expect(cspConfig.scriptSrc).toContain(cdnHost);
+        expect(cspConfig.scriptSrc).not.toContain("'self'");
+        expect(cspConfig.imgSrc).toContain(cdnHost);
+        expect(cspConfig.imgSrc).toContain("'self'");
+        expect(cspConfig.styleSrc).toContain(cdnHost);
+        expect(cspConfig.styleSrc).not.toContain("'self'");
+        expect(cspConfig.connectSrc).toContain(apiHost);
+        expect(cspConfig.connectSrc).not.toContain("'self'");
+      });
+    }
+  });
+
+  describe('CSP defaults', () => {
+    beforeEach(() => {
+      process.env.NODE_ENV = 'production';
+      // Reset the require cache so that the config require
+      // takes into account changes to NODE_ENV.
+      jest.resetModules();
+    });
+
+    it('should default default-src to "\'none\'"', () => {
+      // eslint-disable-next-line global-require
+      const config = require('config');
+      const cspConfig = config.get('CSP').directives;
+      expect(cspConfig.defaultSrc).toEqual(["'none'"]);
+    });
+
+    it('should default base-uri to "\'self\'"', () => {
+      // eslint-disable-next-line global-require
+      const config = require('config');
+      const cspConfig = config.get('CSP').directives;
+      expect(cspConfig.baseUri).toEqual(["'self'"]);
+    });
+
+    it('should default form-action to "\'none\'"', () => {
+      // eslint-disable-next-line global-require
+      const config = require('config');
+      const cspConfig = config.get('CSP').directives;
+      expect(cspConfig.formAction).toEqual(["'none'"]);
+    });
+
+    it('should default frame-src to "\'none\'"', () => {
+      // eslint-disable-next-line global-require
+      const config = require('config');
+      const cspConfig = config.get('CSP').directives;
+      expect(cspConfig.frameSrc).toEqual(["'none'"]);
+    });
+
+    it('should default child-src to "\'none\'"', () => {
+      // eslint-disable-next-line global-require
+      const config = require('config');
+      const cspConfig = config.get('CSP').directives;
+      expect(cspConfig.childSrc).toEqual(["'none'"]);
+    });
+
+    it('should default object-src to "\'none\'"', () => {
+      // eslint-disable-next-line global-require
+      const config = require('config');
+      const cspConfig = config.get('CSP').directives;
+      expect(cspConfig.objectSrc).toEqual(["'none'"]);
+    });
+
+    it('should default media-src to "\'none\'"', () => {
+      // eslint-disable-next-line global-require
+      const config = require('config');
+      const cspConfig = config.get('CSP').directives;
+      expect(cspConfig.mediaSrc).toEqual(["'none'"]);
+    });
+  });
+
   describe('CSP Middleware', () => {
     const existingNodeEnv = process.env.NODE_ENV;
+    const existingNodeAppInstance = process.env.NODE_APP_INSTANCE;
 
     afterEach(() => {
       process.env.NODE_ENV = existingNodeEnv;
-      delete process.env.NODE_APP_INSTANCE;
+      process.env.NODE_APP_INSTANCE = existingNodeAppInstance;
```

Comment:
This change looks good but the `afterEach()` function needs to move just under the top-level `describe(__filename...`; it needs to clean up after *all* the tests in this suite, not just the `CSP Middleware` tests.

Reply:
As per offline convo (on IRC), we'll need to slap in an `if` block to make sure that we actually have `NODE_APP_INSTANCE` in our env vars before restoring it!